### DISCUSSION TITLE
pipenv setting to cache for virtualenv

### DISCUSSION THREAD

**Discussion by digitalnomd** [pypa/pipenv#5645] [Q&A] [answered]
Hello,

I've been using `git worktree` locally and have noticed that my pipenv virtualenv has to be recreated for every single branch. I'm wondering if there's an existing pipenv feature that will enable a cache so that my virtualenv doesn't have to be rebuilt every time? I've tried using `PIPENV_CACHE_DIR` (see [here](https://pipenv.pypa.io/en/latest/configuration/#changing-cache-location)) but this is still a problem.

Thanks in advance!

**Comment by matteius** [comment, marked as answer]
Try setting the https://pipenv.pypa.io/en/latest/virtualenv/#virtual-environment-name -- I suspect the git worktree is confusing pip to recompute the hash, but that setting should give it a constant name to reference.

**Comment by digitalnomd** [comment]
Thanks. That worked @matteius
